File: backend/src/types/reservation.ts
export interface Reservation {
    id: number;
    festival_id: number;
    entity_id: number;
    table_count: number;
    big_table_count: number;
    town_table_count: number;
    note: string | null;
    status: string;
    interactions?: ReservationInteraction[];
    games?: ReservationGame[];
}

export interface ReservationInteraction {
    id: number;
    reservation_id: number;
    description: string | null;
    interaction_date: string;
}

export interface ReservationGame {
    id: number;
    reservation_id: number;
    game_id: number;
    amount: number;
    table_count: number;
    big_table_count: number;
    town_table_count: number;
    status: string;
}

export enum ReservationStatus {
    TO_BE_CONTACTED = "TO_BE_CONTACTED",
    CONTACTED = "CONTACTED",
    IN_DISCUSSION = "IN_DISCUSSION",
    FACTURED = "FACTURED",
    CONFIRMED = "CONFIRMED",
    ABSENT = "ABSENT",
}

export enum ReservationGameStatus {
    ASKED = "ASKED",
    CONFIRMED = "CONFIRMED",
    RECEIVED = "RECEIVED",
    CANCELLED = "CANCELLED",
}

File: androidapp/app/src/main/java/fr/ayae/festivals/data/Reservation.kt
package fr.ayae.festivals.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
@Entity(tableName = "reservations")
data class Reservation(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val festival_id: Int,
    val entity_id: Int,
    val entity_name: String? = null,
    val table_count: Int,
    val big_table_count: Int,
    val town_table_count: Int,
    val electrical_outlets: Int,
    val note: String?,
    val status: String,
    val presented_by_them: Boolean? = false,
    val interactions: List<ReservationInteraction>? = null,
    val games: List<ReservationGame>? = null
)

@Serializable
data class UpdateReservationRequest(
    val table_count: Int? = null,
    val big_table_count: Int? = null,
    val town_table_count: Int? = null,
    val electrical_outlets: Int? = null,
    val note: String? = null,
    val status: String? = null,
    val presented_by_them: Boolean? = null
)

@Serializable
data class UpdateReservationGameRequest(
    val amount: Int? = null,
    val table_count: Int? = null,
    val big_table_count: Int? = null,
    val town_table_count: Int? = null,
    val electrical_outlets: Int? = null,
    val floor_space: Double? = null,
    val status: String? = null
)

@Serializable
data class AddReservationRequest(
    val entity_id: Int,
    val table_count: Int? = null,
    val big_table_count: Int? = null,
    val town_table_count: Int? = null,
    val note: String? = null,
    val status: String? = null,
    val presented_by_them: Boolean? = null
)
